src/commands/Checkin.ts

```diff
@@ -196,37 +196,121 @@ export default class Checkin extends Command {
-      quietZone: 40,
-      title: event.title.substring(0, 36) === event.title ? event.title : event.title.substring(0, 36).concat('...'),
-      titleTop: -20,
-      // Add a subtitle for the literal check-in code as well, so you can read it
-      // if desired without scanning the QR code.
-      subTitle: `Check-in code: ${event.attendanceCode}`,
-      subTitleTop: -5,
-    });
+  private static async generateQRCodeURL(event: PortalEvent, expressCheckinURL: URL, needsSlide: boolean) {
+    // Doesn't need landscape QR slide. Return the QR code by itself
+    let qrCodeDataUrl;
+    if (needsSlide) {
+      const eventQrCode = QR.generateQR(expressCheckinURL.toString(), '', '');
+      qrCodeDataUrl = await this.createQRSlide(event, eventQrCode);
+    } else {
+      const eventQrCode = QR.generateQR(
+        expressCheckinURL.toString(),
+        event.title,
+        `Check-in code: ${event.attendanceCode}`
+      );
+      qrCodeDataUrl = await eventQrCode;
+    }
+
+    return qrCodeDataUrl;
+  }
+
+  /**
+   * Creates a slide with the given QR Code and returns its URL.
+   * @param event Portal Event to create the slide for.
+   * @param eventQrCode QR Code for the event.
+   * @returns URL of the generated slide.
+   */
+  private static async createQRSlide(event: PortalEvent, eventQrCode: string) {
+    // Creating slide with Canvas
+    // Helpful resource: https://blog.logrocket.com/creating-saving-images-node-canvas/
+    const slide = createCanvas(1920, 1080);
+    const context = slide.getContext('2d');
+    // context.fillStyle = '#ffffff';
+    context.fillRect(0, 0, 1920, 1080);
+
+    // draw background
+    const background = await loadImage('./src/assets/qr-slide-background.png');
+    context.drawImage(background, 0, 0, 1920, 1080);
+
+    // draw QR code
+    const angleInRadians = Math.PI / 4;
+    context.rotate(angleInRadians);
+    const qrImg = await loadImage(await eventQrCode);
+    context.drawImage(qrImg, 375, -325, 600, 600);
+    context.rotate(-1 * angleInRadians);
+
+    // members.acmucsd.com
+    // context.textAlign = 'center';
+    // context.font = "40pt 'DM Sans'";
+    // context.fillStyle = '#727272EB';
+    // context.fillText('members.acmucsd.com', 1550, 1000);
+
+    // for
+    // context.lineWidth = 5;
+    // context.beginPath();
+    // context.moveTo(1000, 400);
+    // context.lineTo(1325, 400);
+    // context.stroke();
+    // context.font = "45pt 'DM Sans'";
+    // context.fillStyle = '#000';
+    // context.fillText('for', 1400, 415);
+    // context.beginPath();
+    // context.moveTo(1475, 400);
+    // context.lineTo(1800, 400);
+    // context.stroke();
+
+    // event title
+    const { title } = event;
+    // let title = "123456789012345678901234567...";
+    // let title = "This is 27 Letters Idk Idk?...";
+    let titleSize = Math.max(Math.min(title.length, 70), 8);
+    titleSize = (-2 * titleSize) / 3 + 65;
```

Comment:
I see we reuse this later, can we have this be a helper function / add a description as to why we're doing this

Reply:
we can just write `const rescaleFont = (size) => (-2 * size) / 3 + 65` and put that under the same function since we won't use it anywhere else